app/controllers/records_controller.rb

```diff
@@ -44,13 +46,14 @@ def download_mpl
     if BSON::ObjectId.legal?(params[:format])
       bundle = Bundle.find(BSON::ObjectId.from_string(params[:format]))
 
-      unless File.exist?(bundle.mpl_path)
-        MplDownloadCreateJob.perform_now(bundle.id.to_s)
+      if bundle.mpl_status != :ready
+        flash[:info] = 'This bundle is currently preparing for download.'
+        redirect_to :back
```

Comment:
Do we want to start the prepare job here if it isn't `:ready` or `:building` (e.g. `:unbuilt`)?